### PR TITLE
fix: show honest per-platform agent version status

### PR DESCRIPTION
**Pull request by bokiko**
Phase 3 of approved remediation. Use existing per-architecture release metadata; show Release N, Legacy (unnumbered), or unknown for older hubs. Missing ARM64/legacy platform entries stay visible without crashing. Available is not claimed as trusted. Matches offered build replaces Up to date; unknown running SHA stays unknown; protocol-1 rollback-floor limitation is visible. File timestamp is accurately labeled modification time. No new backend inventory or dependencies. Kimi implemented; Codex reviewed and corrected edge cases. Validation: 93/93 dashboard tests, lint zero errors (3 pre-existing warnings), production build/typecheck; isolated Playwright desktop/mobile fixtures for numbered+legacy, old hub missing platform, withheld update, pause/resume/reload and viewer controls, no runtime errors or page-wide overflow. No real hub APIs were used. No deployment or fleet changes.

---

> [!NOTE]
> **Low Risk**
> Dashboard copy and pure mapping helpers only—no hub API or rollout behavior changes. CI change only adds runner headroom for race tests.
> 
> **Overview**
> The **Versions** page stops overstating what the hub knows: served binaries use **Available** (not **Trusted**), show **Release** metadata when present, and always render per-platform cards—including explicit **Unavailable** placeholders when a build (e.g. Linux ARM64) is not served. Fleet agent status drops **Up to date** for **Matches offered build** only when the hub contract supports it; missing `running_sha` or older hubs get **unknown** labels instead of a green match. Protocol-1 agents get an inline note that they lack the protocol-2 rollback floor.
> 
> Label logic moves into shared `versions-honesty.mjs` with node tests; `AgentBinaryInfo` documents optional `release`. **docs/versions.md** explains the vocabulary and is linked from the docs index.
> 
> **CI** extends the hub race job: job timeout **20→25** minutes and `go test -race` **15m→20m** so the full suite can finish after recent growth (e.g. Windows tests).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cd891e21059440ee000fe3173f8bbd93eff2e83e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: dashboard/src/lib/versions-honesty.mjs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// update_pending = expected != "" && running_sha != expected). The label only
+// refuses a green match when running_sha is absent (older/malformed
+// response). Never a claim about the newest release anywhere.
+export function agentStatusLabel(agent) {
+  if (agent.update_blocked_reason) {
+    return { kind: "blocked", label: agent.update_pending ? "Withheld" : "Unavailable" };
+  }
+  if (agent.update_pending) return { kind: "pending", label: "Update pending" };
+  if (!agent.running_sha) return { kind: "unknown", label: "Version unknown" };
+  return { kind: "current", label: "Matches offered build" };
```
[the remark this answers is not in the document]
Comment (bokiko) [reply dashboard/src/lib/versions-honesty.mjs:76]: Fixed in cd891e2: legacy responses without per-architecture reporting now show Status unknown (older hub), never an inferred green match. Current hubs retain their policy-derived status. Regression covers missing/empty offered SHAs and legacy positive SHAs; 94/94 tests and production build pass. Browser acceptance and CI rerunning.

**Comment by bokiko** [comment]
@codex review — P2 legacy-hub false match is fixed in cd891e2. 94 dashboard tests, production build and four isolated browser scenarios pass; Claude independently verified all callers. Please re-review.
